src/main/java/duke/tasks/TaskUtils.java

```diff
@@ -0,0 +1,67 @@
+package duke.tasks;
+
+import java.time.LocalDate;
+
+/** Utility class for handling things related to Task **/
+public class TaskUtils {
+    /** cached things */
+    public static String DONE_MARKER = "1";
+    public static String NOT_DONE_MARKER = "0";
+    public static String DELIMITER = " | ";
+
+    /**
+     *
+     * Returns a task encoded by the given String
+     *
+     * @param str
+     * @return Task specified by str
+     */
+    public static Task stringToTask(String str) {
+        String[] segments = str.split(" \\| ");
+        if (segments.length < 3) {
+            return null;
+        }
+        boolean isDone = segments[1].equals(DONE_MARKER);
+        Task toReturn;
+        switch (segments[0]) {
+            case "T":
+                toReturn = new Todo(segments[2]);
+                break;
```

Comment:
Perhaps remove the indentation in the switch statement, as there should be no indentation for `case` clauses. 

One useful way is to configure Intellij's settings so they can handle this indentation for you. Go to `Settings > Editor > Code Style > Java`, select tab `Wrapping and Braces`, scroll down the list to `switch` statement, uncheck the option `Indent 'case' branches`.

```suggestion
        switch (segments[0]) {
        case "T":
            toReturn = new Todo(segments[2]);
            break;
```